include/gsl/span

```diff
@@ -451,45 +454,47 @@ public:
     constexpr span& operator=(const span& other) noexcept = default;
 
     // [span.sub], span subviews
-    template <std::ptrdiff_t Count>
+    template <std::size_t Count>
     constexpr span<element_type, Count> first() const noexcept
     {
-        Expects(Count >= 0 && Count <= size());
+        Expects(Count <= size());
         return {data(), Count};
     }
 
-    template <std::ptrdiff_t Count>
-    GSL_SUPPRESS(bounds.1) // NO-FORMAT: attribute
+    template <std::size_t Count>
+    GSL_SUPPRESS(bounds .1) // NO-FORMAT: attribute
     constexpr span<element_type, Count> last() const noexcept
     {
-        Expects(Count >= 0 && size() - Count >= 0);
+        Expects(size() >= Count);
         return {data() + (size() - Count), Count};
     }
 
-    template <std::ptrdiff_t Offset, std::ptrdiff_t Count = dynamic_extent>
-    GSL_SUPPRESS(bounds.1) // NO-FORMAT: attribute
+    template <std::size_t Offset, std::size_t Count = dynamic_extent>
+    GSL_SUPPRESS(bounds .1) // NO-FORMAT: attribute
     constexpr auto subspan() const noexcept ->
         typename details::calculate_subspan_type<ElementType, Extent, Offset, Count>::type
     {
-        Expects((Offset >= 0 && size() - Offset >= 0) &&
-                (Count == dynamic_extent || (Count >= 0 && Offset + Count <= size())));
+        Expects((size() >= Offset) &&
+                (Count == dynamic_extent || (Offset + Count <= size())));
```

Comment:
`Count` could be so enormous that `Offset + Count` wraps and is `< size()` despite that `Count > size()` (See https://cplusplus.github.io/LWG/issue3103).
```suggestion
                (Count == dynamic_extent || (Count <= size() - Offset)));
```